R/ui_functions.R

```diff
@@ -612,3 +612,110 @@ ui_cp_download_single <- function(country,
 
   return(out)
 }
+
+#' Country Profile Key Indicators download
+#'
+#' Helper function to download Country Profile data
+#'
+#' @inheritParams ui_cp_download
+#' @return list
+#' @keywords internal
+ui_cp_download_multiple <- function(country,
+                                    povline = 1.9,
+                                    lkup,
+                                    indicators = "all") {
+
+
+  ### get indicator list of interest
+  ind_table <- pip(country = country,
+                   povline = povline,
+                   lkup = lkup,
+                   welfare_type = "all",
+                   reporting_level = "all")
+
+  ind_table <- ind_table[,c("country_code", "reporting_year",
+                            "headcount")]
+
+  ##NOTICE THAT THIS USES cp_lkups INSTEAD OF cp. I am using version
+  ##`20220909_2017_01_02_PROD` please check before running
+  # lkup$cp_lkups$key_indicators$shared_prosperity <- NULL
+
+  dl <- lkup$cp_lkups$key_indicators
+
+  ### include all the mpm and sp data as well
+  dl$mpm_headcount <- lkup$cp_lkups$charts$mpm[, c("country_code",
+                                                   "reporting_year",
+                                                   "mpm_headcount")]
+
+  sp_dt <- lkup$cp_lkups$charts$sp[,c("country_code",
+                                      "year_range",
+                                      "distribution",
+                                      "shared_prosperity")]
+
+  sp_dt <- dcast(sp_dt,
+                 country_code + year_range ~ distribution,
+                 value.var = "shared_prosperity",
+                 fun.aggregate = mean)
```

Comment:
Why do we need aggregation here? Could this be moved to the data pipeline stage? I'm thinking about performance here, every computation or transformation that can be done before, should be done at an earlier stage (data pipeline)